### ISSUE TITLE
`mypy` and `dmypy run` are not agreeing

### ISSUE THREAD

**Issue by alanhdu** [python/mypy#15214]
**Bug Report**

I'm not sure if I misunderstood the documentation, but I assumed that `mypy` and `dmypy run` would should always return the same thing when run on the same code. This does *not* seem to be the case though:

**To Reproduce**

If I have the simple file:
```python
_var = None

def set_var():
    global _var
    if not _var:
        _var = 1
    return _var
```

then `mypy foo.py` gives success, while `dmypy run -- foo.py` reports:
```
Daemon started
mypy foo.py
foo.py:3: error: Need type annotation for "_var" (hint: "_var: Optional[<type>] = ...")  [var-annotated]
Found 1 error in 1 file (checked 44 source files)
```

This is on both `mypy 1.2.0 (compiled: yes)` and `mypy 1.4.0+dev.35acb1bf69c990fcd89eeec32e35ae21f93c2d84 (compiled: no)`.

**Expected Behavior**

I would've expected either both of them to fail with the same error, or both of them to succeed. Alternatively, I would've expected the docs to spell out what the configuration differences are between `mypy` and `dmypy`

**Comment by hauntsaninja** [comment]
Yeah, see https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-local-partial-types

**Comment by alanhdu** [comment]
Ah -- of course, I must have missed that when reading the docs. Would you be open to adding a note in https://mypy.readthedocs.io/en/stable/mypy_daemon.html to make that clearer (I'd be happy to open a PR), or is it going to be enabled by default soon enough that we shouldn't bother?

**Comment by hauntsaninja** [comment]
Yes, a note sounds good! I'm not sure there's a concrete timeline towards making it the default (last time it was consciously thought about was to not do it as part of the 1.0 release)
